pinot-broker/src/main/java/org/apache/pinot/broker/routing/RoutingManager.java

```diff
@@ -303,11 +302,11 @@ public synchronized void buildRouting(String tableNameWithType) {
     TableConfig tableConfig = ZKMetadataProvider.getTableConfig(_propertyStore, tableNameWithType);
     Preconditions.checkState(tableConfig != null, "Failed to find table config for table: %s", tableNameWithType);
 
-    IdealState idealState = getIdealState(tableNameWithType);
+    IdealState idealState = getIdealState(getIdealStatePath(tableNameWithType));
```

Comment:
Shall we cache the `idealStatePath` and `externalViewPath` so that they can be reused to create the `RoutingEntry`?

Reply:
I don't think that's wise because this is called by an external API and could be used to OOM the server, unless you have an eviction policy, at which point this is getting far too complicated. The purpose of this change is just to reduce allocation when the broker is ideal and just handling cluster updates.

Reply:
What I meant is to put it in a local variable, and reuse it in line 382 when creating the `RoutingEntry`

Reply:
Oh I see, makes sense. Done.